vector/src/main/java/im/vector/util/ThemeUtils.kt

```diff
@@ -0,0 +1,214 @@
+            THEME_BLACK_VALUE -> activity.setTheme(otherThemes.second)
+        }
+
+        mColorByAttr.clear()
+    }
+
+    /**
+     * Set the TabLayout colors.
+     * It seems that there is no proper way to manage it with the manifest file.
+     *
+     * @param activity the activity
+     * @param layout   the layout
+     */
+    fun setTabLayoutTheme(activity: Activity, layout: TabLayout) {
+        if (activity is VectorGroupDetailsActivity) {
+            val textColor: Int
+            val underlineColor: Int
+            val backgroundColor: Int
+
+            if (TextUtils.equals(getApplicationTheme(activity), THEME_LIGHT_VALUE)) {
+                textColor = ContextCompat.getColor(activity, android.R.color.white)
+                underlineColor = textColor
+                backgroundColor = ContextCompat.getColor(activity, R.color.tab_groups)
+            } else {
+                textColor = ContextCompat.getColor(activity, R.color.tab_groups)
+                underlineColor = textColor
+                backgroundColor = getColor(activity, R.attr.primary_color)
+            }
+
+            layout.setTabTextColors(textColor, textColor)
+            layout.setSelectedTabIndicatorColor(underlineColor)
+            layout.setBackgroundColor(backgroundColor)
+        }
+    }
+
+    /**
+     * Translates color attributes to colors
+     *
+     * @param c              Context
+     * @param colorAttribute Color Attribute
+     * @return Requested Color
+     */
+    @ColorInt
+    fun getColor(c: Context, @AttrRes colorAttribute: Int): Int {
+        if (mColorByAttr.containsKey(colorAttribute)) {
+            return mColorByAttr[colorAttribute] as Int
+        }
+
+        var matchedColor: Int
+
+        try {
+            val color = TypedValue()
+            c.theme.resolveAttribute(colorAttribute, color, true)
+            matchedColor = color.data
+        } catch (e: Exception) {
+            matchedColor = ContextCompat.getColor(c, android.R.color.holo_red_dark)
+        }
+
+        mColorByAttr[colorAttribute] = matchedColor
+
+        return matchedColor
+    }
+
+    /**
+     * Get the resource Id applied to the current theme
+     *
+     * @param c          the context
+     * @param resourceId the resource id
+     * @return the resource Id for the current theme
+     */
+    fun getResourceId(c: Context, resourceId: Int): Int {
+        if (TextUtils.equals(getApplicationTheme(c), THEME_DARK_VALUE)) {
+
+            if (resourceId == R.drawable.line_divider_light) {
+                return R.drawable.line_divider_dark
+            }
+        }
+        return resourceId
+    }
+
+    /**
+     * Update the menu icons colors
+     *
+     * @param menu  the menu
+     * @param color the color
+     */
+    fun tintMenuIcons(menu: Menu, color: Int) {
+        for (i in 0 until menu.size()) {
+            val item = menu.getItem(i)
+            val drawable = item.icon
```

Comment:
Can we consider here a potential item.title?
(see https://github.com/dinsic-pim/riot-android/pull/172/files#diff-d2c1df037c6ac766c6370a262f602269R2129)

Reply:
As discussed with @Tyuoli , it is ok like that